### DISCUSSION TITLE
show_var_residual for variables of type LAGRANGE_VEC

### DISCUSSION THREAD

**Discussion by raj-anant** [idaholab/moose#24817] [Q&A General]
Dear MOOSE Team,

We are using MALAMUTE for melt pool simulations and needed to study how the residuals vary in response to random perturbations to the solution. We are using show_var_residual in the Debug block for dumping the residuals to the Exodus file. It works well for scalar variables like temperature and pressure but gives error for vector variables like velocity. For instance, if we use the following:
```
[Variables]
  [velocity]
    family = LAGRANGE_VEC
  []
[]
[Debug]
  show_var_residual =  'velocity'
[]
```
It gives the error: "No standard variable named velocity found. Did you specify a vector variable when you meant to specify a standard variable"

We tried using specific components like show_var_residual =  'velocity_x', but that leads to a segmentation fault.

Is there a workaround for this issue? We would appreciate any help on this.

Thanks,

Anant

**Comment by GiudGiud** [comment]
Hello

We have no implemented variable residual debugging for vector variables.
You ll want to modify the DebugResidualAux to make it work for vector variables

Guillaume
